File: Generated_Codes_Check/Results/java/225.java
/*

Explanation
Test Case 1:
The given map is as follows



Test Case 2:
The given map is as follows

There are three CPCs in the map.

CPC 1:

L = 5
R = 4
T = 5
B = 3
Possible primes are 2, 3, 5.

CPC 2:

L = 2
R = 2
T = 2
B = 2
Possible primes are 2.

CPC 3:

L = 1
R = 1
T = 1
B = 1
Possible primes are 2.

Therefore, there are 3 monsters in the map.
*/

import java.util.*;

public class CodeArtOnline_Stage2 {
	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		int t = s.nextInt();
		for (int x = 0; x < t; x++) {
			int r = s.nextInt();
			int c = s.nextInt();
			int[][] array = new int[r][c];
			char[][] input = new char[r][c];
			for (int i = 0; i < r; i++) {
				for (int j = 0; j < c; j++) {
					input[i][j] = s.next().charAt(0);
					if (input[i][j] == '#') {
						array[i][j] = -1;
					}
				}
			}

			for (int i = 0; i < r; i++) {
				for (int j = 0; j < c; j++) {
					if (input[i][j] == '^') {
						int left = findLeft(array, i, j);
						int right = findRight(array, i, j);
						int top = findTop(array, i, j);
						int bottom = findBottom(array, i, j);

						int min = Math.min(left, right);
						min = Math.min(min, top);
						min = Math.min(min, bottom);

						if (min != 0) {
							if (checkPrime(min)) {
								array[i][j] = 1;
							}
						}
					}
				}
			}
			int count = 0;
			for (int i = 0; i < r; i++) {
				for (int j = 0; j < c; j++) {
					if (array[i][j] == 1) {
						count++;
					}
				}
			}
			System.out.println(count);
		}
	}

	static public int findLeft(int[][] array, int i, int j) {
		int count = 0;
		for (int x = j; x >= 0; x--) {
			if (array[i][x] == -1) {
				break;
			} else {
				count++;
			}
		}
		return count;
	}

	static public int findRight(int[][] array, int i, int j) {
		int count = 0;
		for (int x = j; x < array[0].length; x++) {
			if (array[i][x] == -1) {
				break;
			} else {
				count++;
			}
		}
		return count;
	}

	static public int findTop(int[][] array, int i, int j) {
		int count = 0;
		for (int x = i; x >= 0; x--) {
			if (array[x][j] == -1) {
				break;
			} else {
				count++;
			}
		}
		return count;
	}

	static public int findBottom(int[][] array, int i, int j) {
		int count = 0;
		for (int x = i; x < array.length; x++) {
			if (array[x][j] == -1) {
				break;
			} else {
				count++;
			}
		}
		return count;
	}

	static public boolean checkPrime(int x) {
		for (int i = 2; i <= x / 2; i++) {
			if (x % i == 0) {
				return false;
			}
		}
		return true;
	}
}

File: Generated_Codes_Check/Results/python/225.py
"""


"""

#Solution 1 (Recursion)
import math
def check_prime(num):
    if num > 1:
        for i in range(2, int(math.sqrt(num)) + 1):
            if num % i == 0:
                return False
        return True
    return False

def check_cpc(grid, i, j, r, c):
    if grid[i][j] == '#':
        return 0
    left = 1
    right = 1
    top = 1
    bottom = 1
    for l in range(j - 1, -1, -1):
        if grid[i][l] == '#':
            break
        left += 1
    for r in range(j + 1, c):
        if grid[i][r] == '#':
            break
        right += 1
    for t in range(i - 1, -1, -1):
        if grid[t][j] == '#':
            break
        top += 1
    for b in range(i + 1, r):
        if grid[b][j] == '#':
            break
        bottom += 1
    min_val = min(left, right, top, bottom)
    count = 0
    for n in range(2, min_val + 1):
        if check_prime(n):
            count += 1
    return count

def get_monster_count(grid, r, c):
    count = 0
    for i in range(r):
        for j in range(c):
            count += check_cpc(grid, i, j, r, c)
    return count

grid = [
        ['#', '^', '^', '^', '^', '^', '#', '^', '#', '#'],
        ['^', '^', '^', '^', '^', '^', '^', '^', '^', '^'],
        ['^', '^', '^', '^', '^', '^', '^', '^', '^', '#']
        ]
r = 3
c = 10
print(get_monster_count(grid, r, c))

grid = [
        ['^', '^', '^', '^', '#', '^', '^', '^', '^', '#'],
        ['^', '^', '^', '^', '^', '^', '^', '#', '^', '^'],
        ['^', '^', '#', '#', '^', '^', '#', '^', '#', '^'],
        ['#', '^', '^', '^', '^', '^', '^', '^', '#', '^'],
        ['^', '^', '#', '^', '^', '^', '^', '^', '^', '^'],
        ['^', '^', '^', '^', '#', '^', '^', '^', '^', '^'],
        ['^', '^', '^', '^', '^', '^', '^', '^', '^', '^']
        ]
r = 7
c = 10
print(get_monster_count(grid, r, c))


#Solution 2 (Non-Recursion)